### PR TITLE
Get current layer instance for TileLayer

### LINKED ISSUE #4893: Ability to pass additional context to `getTileData`
#### Target Use case

Sometimes the implementation of `getTileData` changes, but in a way that only affects getting additional tiles. The current way to inform `TilLayer` that your `getTileData` changed is to pass:

```
new TileLayer({
  getTileData(tile) {
    // do something with tile and somethingYourGetTileDataFuncIsDependentOn
  },
  updateTriggers: {
    getTileData: somethingYourGetTileDataFuncIsDependentOn
  }
})
```

However, when using this approach, `TileLayer` will re-request all the tiles.

#### Proposed feature

Passing the current layer instance to `getTileData` provides a way to give different props to `TileLayer` and have those props passed to `getTileData` for subsequent tile requests without needing to re-request the already-loaded tiles.

With this approach you can accomplish the same like this:

```
new TileLayer({
  getTileData(tile, layer) {
    // do something with tile and layer.props.somethingYourGetTileDataFuncIsDependentOn
  },
  somethingYourGetTileDataFuncIsDependentOn,
})
```

#### To Do List 
  - [ ] Add label and assign to milestone 
  - [ ] Coding
  - [ ] Doc update
  - [ ] What’s new update 
  - [ ] Test

### PR DESCRIPTION
**Pull request by supersonicclay**
For #4893

#### Change List
-

Fixes #4893

### CODE REVIEW

**Comment by Pessimistress** [comment]
Theoretically, when props like `data` and `elevationDecoder` change, all tiles should be invalidated and reload.

Let's limit this PR to using the current `getTileData`.

**Comment by supersonicclay** [comment]
Done @Pessimistress

**Comment by kylebarron** [comment]
So instead of passing the `layer` instance, this just uses the "current" value of `getTileData`? Would be good to add a line in What's New

**Comment by supersonicclay** [comment]
Done @kylebarron

**Comment by supersonicclay** [comment]
@kylebarron @Pessimistress is there anything else you need with this or can we get it merged? Thanks!

File: modules/geo-layers/src/tile-layer/tile-layer.js [hunk 1]
```diff
@@ -143,7 +143,8 @@ export default class TileLayer extends CompositeLayer {
   // Methods for subclass to override
 
   getTileData(tile) {
-    const {getTileData, fetch, data} = this.props;
+    const {fetch, data} = this.props;
```
Comment (Pessimistress) [inline modules/geo-layers/src/tile-layer/tile-layer.js:146]: `fetch` should be taken from the current props too

**Comment by supersonicclay** [comment]
Done @Pessimistress

**Pull request merged**